graylog2-web-interface/src/components/users/EditRolesForm.jsx

```diff
@@ -39,11 +43,22 @@ const EditRolesForm = React.createClass({
   _onCancel() {
     this.props.history.pushState(null, Routes.SYSTEM.AUTHENTICATION.USERS.LIST);
   },
+  _onValueChange(newRoles) {
+    const roles = newRoles.split(',');
+    this.setState({ newRoles: roles });
+  },
   render() {
     const user = this.props.user;
     if (!this.state.roles) {
       return <Spinner />;
     }
+    let rolesAlert = null;
+    const roles = this.state.newRoles;
+    if (roles != null && roles.indexOf('Reader') === -1 && roles.indexOf('Admin') === -1) {
```

Comment:
You could replace `roles.indexOf()` with [`roles.includes()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes) for better reading experience. The polyfill we use should take care of converting it if the browser does not support it.